components/script/dom/globalscope.rs

```diff
@@ -3310,6 +3312,28 @@ impl GlobalScope {
         }
         unreachable!();
     }
+
+    pub(crate) fn report_csp_violations(&self, violations: Vec<Violation>) {
+        for violation in violations {
+            let sample = match violation.resource {
+                ViolationResource::Inline { .. } | ViolationResource::Url(_) => None,
+                ViolationResource::TrustedTypePolicy { sample } => Some(sample),
+            };
+            let report = CSPViolationReportBuilder::default()
+                .resource("eval".to_owned())
```

Comment:
Why is the [violation resource](https://www.w3.org/TR/CSP3/#violation-resource) always "eval"? For example, when blocking a fetch request then it should be the requests url right?

Reply:
This was implemented before (https://github.com/servo/servo/blob/3babf7498656b9ff41b9d7894849a1921c68f28f/components/script/dom/globalscope.rs#L2679) and I think this is a placeholder. It was moved yesterday in #36493 where it was always set before in the class itself: https://github.com/servo/servo/blob/70c0faa0e9b4b1babcbf3e0c84b1213c58bb4167/components/script/security_manager.rs#L86

We should fix that, but I think it is unrelated to the current refactoring (and this PR is already quite big).